### DISCUSSION TITLE
Is it possible to drop distlib dependency?

### DISCUSSION THREAD

**Discussion by frostming** [pypa/virtualenv#2473] [Ideas]
From the [search result](https://github.com/search?q=repo%3Apypa%2Fvirtualenv+distlib&type=code) it seems `distlib` isn't critical and directly depended by the source code. So I am asking whether it is possible to drop it from the dependencies. It will reduce the package size of virtualenv since `distlib` is a rather large dependency.

**Comment by gaborbernat** [comment]
It's very critical - https://github.com/pypa/virtualenv/blob/main/src/virtualenv/seed/embed/via_app_data/pip_install/base.py#L11 Your search query failed you there.

**Reply by frostming** [reply]
Ah yes, the search result misguided me.

**Reply by gaborbernat** [reply]
I don't think there's anything better as a replacement.

**Comment by gaborbernat** [comment]
This discussion has been answered. The `distlib` dependency is critical to virtualenv's functionality and is directly used in `src/virtualenv/seed/embed/via_app_data/pip_install/base.py`, so it cannot be dropped. Closing as answered.
